[ao][sparsity][fx] make sparse prepare->quant prepare compose

HDCharles opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #82204
* #81994
* __->__ #81993
* #81992

Summary: The primary issue was that fusion and matching had to be
updated to handle parametrized modules

Test Plan: python test/test_ao_sparsity.py TestFxComposability

Reviewers:

Subscribers:

Tasks:

Tags:

HDCharles committed:
[ao][sparsity][fx] make sparse prepare->quant prepare compose

Summary: The primary issue was that fusion and matching had to be
updated to handle parametrized modules

Test Plan: python test/test_ao_sparsity.py TestFxComposability

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity][fx] make sparse prepare->quant prepare compose"


Summary: The primary issue was that fusion and matching had to be
updated to handle parametrized modules

Test Plan: python test/test_ao_sparsity.py TestFxComposability

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on test/ao/sparsity/test_composability.py:
```diff
@@ -376,3 +376,43 @@ def test_q_prep_fx_before_s_prep(self):
         )
         self.assertGreaterAlmostEqual(cur_sparsity, sparse_config[0]["sparsity_level"])
+
+    def test_s_prep_before_q_prep_fx(self):
```
can we add a docstring describing at a high level what we are testing for this test as well

HDCharles replied:
yeah, will do

HDCharles committed:
Update on "[ao][sparsity][fx] make sparse prepare->quant prepare compose"


Summary: The primary issue was that fusion and matching had to be
updated to handle parametrized modules

Test Plan: python test/test_ao_sparsity.py TestFxComposability

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity][fx] make sparse prepare->quant prepare compose"


Summary: The primary issue was that fusion and matching had to be
updated to handle parametrized modules

Test Plan: python test/test_ao_sparsity.py TestFxComposability

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles commented:
@pytorchmergebot merge -g
